OSSM-9365: Resolve documentations issues for OSSM from OSDDOCs (#94730)
openshift/openshift-docs: modules/ossm-accessing-bookinfo-application-using-gateway-api.adoc

:_mod-docs-content-type: Procedure
[id="ossm-accessing-bookinfo-application-using-gateway-api"]
= Accessing the Bookinfo application by using Gateway API

> The {k8s} Gateway API deploys a gateway by creating a `Gateway` resource. In {ocp-product-title} 4.15 and later, the Gateway API custom resource definitions (CRDs) are available but disabled by default. To use them, you must manually enable the CRDs. In versions 4.18 and earlier, you must manually install the CRDs. Starting with {ocp-product-title} 4.19, the platform automatically installs and manages these CRDs, and you can no longer create, update, or delete them.

Reviewer: It seems confusing to me to talk about "enabling" the CRDs on OCP 4.15+ and then "installing" the CRDs on 4.18 and earlier.  

OCP 4.15 is the first version on which OSSM supports Gateway API, if I understand correctly, and OCP 4.19 is the first version on which OCP manages the CRDs.  
```suggestion
The {k8s} Gateway API deploys a gateway by creating a `Gateway` resource. In {ocp-product-title} 4.15 and later, {SMProductName} implements the Gateway API custom resource definitions (CRDs).  However, the CRDs themselves are not installed by default in {ocp-product-title} 4.18 and earlier. This means that in {ocp-product-title} 4.15 through 4.18, you must manually install the CRDs. Starting with {ocp-product-title} 4.19, the platform automatically installs and manages these CRDs, and you can no longer create, update, or delete them.
```

.Procedure

. Enable the Gateway API CRDs for {ocp-product-title} 4.18 and earlier, by running the following command:
+
[source,terminal]
----
$ oc get crd gateways.gateway.networking.k8s.io &> /dev/null ||  { oc kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v1.0.0" | oc apply -f -; }
----

> . Create and configure a gateway using a `Gateway` resource and `HTTPRoute` resource by running the following command:

Reviewer 2: ```suggestion
. Create and configure a gateway by using the `Gateway` and `HTTPRoute` resources by running the following command:
```

To configure a gateway with the `bookinfo` application by using the Gateway API, this example uses a sample gateway configuration file that must be applied in the namespace where the application is installed.
====

. Ensure that the Gateway API service is ready, and has an address allocated by running the following command:
+
[source,terminal]
----
$ oc wait --for=condition=programmed gtw bookinfo-gateway -n bookinfo
----

. Retrieve the host by running the following command:
+
[source,terminal]
----
$ export INGRESS_HOST=$(oc get gtw bookinfo-gateway -n bookinfo -o jsonpath='{.status.addresses[0].value}')

> ----
> . Retrieve the port by running the following command:
> +

Reviewer 2: ```suggestion
----

. Retrieve the port by running the following command:
+
```

Add a new line

[source,terminal]
----
$ export INGRESS_PORT=$(oc get gtw bookinfo-gateway -n bookinfo -o jsonpath='{.spec.listeners[?(@.name=="http")].port}')
----

. Retrieve the gateway URL by running the following command:
+
[source,terminal]
----
$ export GATEWAY_URL=$INGRESS_HOST:$INGRESS_PORT
----

. Obtain the gateway host name and the URL of the product page by running the following command:
+
[source,terminal]
----
$ echo "http://${GATEWAY_URL}/productpage"
----

.Verification

> . Verify that the `productpage` is accessible from a web browser.

Reviewer 2: ```suggestion
* Verify that the *productpage* is accessible from a web browser.
```

- has a single step hence use `*`. [guideline](https://redhat-documentation.github.io/supplementary-style-guide/#single-step-procedures)
- https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference
> Use single asterisks for web console/GUI items....
